src/CMakeLists.txt

```diff
@@ -4,84 +4,84 @@ find_package(JNI REQUIRED)
+        ${HEADERS_DIR_CPPJNI}/jni_wrappers/generate_jni_signature_string.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_wrappers/global_reference.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_wrappers/global_reference.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_wrappers/local_reference.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_wrappers/local_reference.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_wrappers/method_id_cache.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/generate_overloaded_constructor_declarations.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/generate_overloaded_method_declarations.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/jni_constructor_signature_hack.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/method_declaration.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/method_scope_category.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/scoped_declaration.hpp
+        ${HEADERS_DIR_CPPJNI}/meta_declaration/signature_info.hpp
+        ${HEADERS_DIR_CPPJNI}/method_call/method_caller.hpp
+        ${HEADERS_DIR_CPPJNI}/mpl/list.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/lang/class.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/lang/class.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/lang/comparable.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/lang/object.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/lang/string.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/lang/throwable.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/util/iterator.hpp
+        ${HEADERS_DIR_CPPJNI}/predefined/java/util/list.hpp
+        ${HEADERS_DIR_CPPJNI}/types/base_object_container.hpp
+        ${HEADERS_DIR_CPPJNI}/types/class.hpp
+        ${HEADERS_DIR_CPPJNI}/types/object.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/types/object.hpp
+        ${HEADERS_DIR_CPPJNI}/types/object_converter.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/types/object_converter.hpp
+        ${HEADERS_DIR_CPPJNI}/types/primitives.hpp
+        ${HEADERS_DIR_CPPJNI}/types/string.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/types/string.hpp
+        ${HEADERS_DIR_CPPJNI}/java_cast.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/java_cast.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_exception.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_thread_hint.hpp
+        ${HEADERS_DIR_CPPJNI}/jni_version.hpp
+        ${HEADERS_DIR_CPPJNI}/method_signature.hpp
+        ${HEADERS_DIR_CPPJNI}/native_cast.fwd.hpp
+        ${HEADERS_DIR_CPPJNI}/native_cast.hpp
+        ${HEADERS_DIR_CPPJNI}/virtual_machine.hpp
+        ${HEADERS_DIR_CPPJNI}/virtual_machine.fwd.hpp
+        )
 
 add_library(${PROJECT_NAME} SHARED ${SOURCES} ${HEADERS})
 
 target_include_directories(
-    ${PROJECT_NAME}
-    PUBLIC
+        ${PROJECT_NAME}
+        PUBLIC
         ${HEADERS_DIR}
         ${JNI_INCLUDE_DIRS}
-        ${Boost_INCLUDE_DIRS}
 )
 
 target_link_libraries(${PROJECT_NAME}
-    ${JNI_LIBRARIES}
-)
\ No newline at end of file
+        ${JNI_LIBRARIES}
+        hana
```

Comment:
Doesn't hana package contain some sort of CMake variable containing library names? It would be the best to use that instead of handwritten name.

Reply:
this is imported target, generated by CMake